### ISSUE TITLE
False positive `reportArgumentType` when using walrus operator in match statement

### ISSUE THREAD

**Issue by Apakottur** [microsoft/pyright#11066]
**Describe the bug**
Consider the following code:
```python
# test.py
def inner(val: int) -> None:
    print(val)

def outer() -> None:
    response_json: dict[str, int | float] = {"code": 1, "value": 3.14}

    match error_code := response_json["code"]:
        case 1:
            inner(error_code)

outer()
```

Running `pyright test.py` gives:
```text
./test.py
  ./test.py:11:19 - error: Argument of type "float | Literal[1]" cannot be assigned to parameter "val" of type "int" in function "inner"
    Type "float | Literal[1]" is not assignable to type "int"
      "float" is not assignable to "int" (reportArgumentType)
1 error, 0 warnings, 0 informations
```

The code is valid and prints "1" as expected, but Pyright does not correctly infer the argument type as `Literal[1]`.
Am I missing something or is this a bug?

I'm using `pyright==1.1.407`.

**Comment by erictraut** [comment]
This isn't specific to a walrus operator, and it's not a bug. 

Literal patterns in a `match` statement use equality tests, so a value of `1.0` (a float) will match against a literal pattern of `1`. You can conform this by running the following code:

```python
def test(val: float) -> None:
    match val:
        case 1:
            print("got 1")

test(1.0)
```

It would therefore be incorrect for pyright to narrow the type of `error_code` to `Literal[1]` in your code sample. It correctly narrows it to `float | Literal[1]`.
